pkg/controller/dataimportcron-controller.go

```diff
@@ -1088,6 +1087,95 @@ func addDataImportCronControllerWatches(mgr manager.Manager, c controller.Contro
 		return err
 	}
 
+	if err := addDefaultStorageClassUpdateWatch(mgr, c); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// addDefaultStorageClassUpdateWatch watches for default/virt default storage class updates
+func addDefaultStorageClassUpdateWatch(mgr manager.Manager, c controller.Controller) error {
+	if err := c.Watch(source.Kind(mgr.GetCache(), &storagev1.StorageClass{}),
+		handler.EnqueueRequestsFromMapFunc(
+			func(ctx context.Context, obj client.Object) []reconcile.Request {
+				log := c.GetLogger().WithName("StorageClassWatch")
+				log.Info("Default/default virt SC updated", "scName", obj.GetName())
+				if err := deleteOutdatedPendingPvcs(ctx, mgr.GetClient(), log); err != nil {
+					log.Error(err, "Failed deleting outdated pending PVCs")
```

Comment:
On the other hand this may be a good candidate for "fire once and never retry" so not sure

Reply:
We may consider instead to reconcile all the DICs whose latest import PVC is pending on default SC, and delete it from the reconcile itself.

Reply:
I see you already took the reconciler approach, but maybe the initial one made more sense, tough call
Downside of having this in reconciler is increased chance of a bug in the form of infinite loop of PVC deletes

> On the other hand this may be a good candidate for "fire once and never retry" so not sure

Reply:
The reconciler approach seems more k8s-oriented. Why should we be afraid of an infinite PVC delete loop but not from the already possible infinite PVC create loop?

Reply:
As far as I can tell it's because we'll just ignore the already exists error. Yes this is definitely the more expected approach